frontend/packages/console-shared/src/components/editor/yaml-editor-utils.ts

```diff
@@ -1,131 +1,15 @@
-import * as URL from 'url';
-import { Uri, Range } from 'monaco-editor';
-import {
-  MonacoToProtocolConverter,
-  ProtocolToMonacoConverter,
-} from 'monaco-languageclient/lib/monaco-converter';
+import { Range } from 'monaco-editor';
+import type * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
+import { configureMonacoYaml } from 'monaco-yaml';
 import * as yaml from 'yaml-ast-parser';
-import { getLanguageService, TextDocument } from 'yaml-language-server';
 import { openAPItoJSONSchema } from '@console/internal/module/k8s/openapi-to-json-schema';
 import { getSwaggerDefinitions } from '@console/internal/module/k8s/swagger';
 
 export const defaultEditorOptions = { readOnly: false, scrollBeyondLastLine: false };
 
-const MODEL_URI = 'inmemory://model.yaml';
```

Comment:
Love seeing all of this code removed

Reply:
Hey! Yeah, it would likely impact plugins since the monaco API has changed a bit between versions. I've also changed how the ref object is stored in memory, I'll try making the ref backwards-compatible and will push a change if successful